share/functions/__fish_print_hostnames.fish

```diff
@@ -59,17 +59,25 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
                 # Normalize whitespace
                 | string trim | string replace -r -a '\s+' ' ')
             end
-            if test -n "$paths"
-                # Expand paths which may have globbing and tokenize
-                set paths (eval "echo $paths" | string split ' ')
-                for path_index in (seq (count $paths))
-                    # Resolve relative paths
-                    if string match -v '/*' $paths[$path_index] >/dev/null
-                        set paths[$path_index] $relative_path/$paths[$path_index]
+
+            set -l new_paths
+            for path in $paths
+                # Guard against echo command complaining on zero wildcard matches
+                if test (eval "count $path") -gt 0
```

Comment:
Use `set` to inhibit the glob match failure:

```
set -l expanded_path
eval set expanded_path (echo $path)
for path in $expanded_path
```

This is a good example for why we should implement the proposed `glob` command.